clients/notification/src/main/java/org/eclipse/hono/notification/deviceregistry/DevicesOfTenantDeletedNotification.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.notification.deviceregistry;
+
+import java.time.Instant;
+import java.util.Objects;
+
+import org.eclipse.hono.annotation.HonoTimestamp;
+import org.eclipse.hono.notification.AbstractNotification;
+import org.eclipse.hono.notification.NotificationConstants;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Notification that informs that all devices of a tenant have been deleted.
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class DevicesOfTenantDeletedNotification extends AbstractNotification {
```

Comment:
IMHO naming to `AllDevicesOfTenantDeletedNotification` would be more suited as all the devices belonging to a tenant  are being deleted.